### DISCUSSION TITLE
HomeScreen: Open a folder by default

### DISCUSSION THREAD

**Discussion by mipro98** [GitJournal/GitJournal#818] [Feature Requests]
It would be cool if there was an option to specify a default folder which opens when starting the app.

For example when you use GitJournal for both notes and as a journal, the default "Root Folder" has all notes and journal files mixed which might not be wanted.
#94 could help a bit but it would be awesome if a custom folder opens up as default.

**Comment by vHanda** [comment]
So my current plan is -

* Allow you to set a custom default folder for the journal editor.
* Allow each folder to have its own settings (already implemented but needs some UI work to expose it better). This way the Journal folder can have a separate view, and filename and sorting settings.

* Allow some way to change the default home screen. Instead of All Notes, allow any folder or the Folders View.

I think with these (3) things and eventual folder pinning, it should be good, right? Or can you think of something else?

**Comment by mipro98** [comment]
This sounds awesome - especially because the folder-specific settings has been another idea i had.

**Comment by vHanda** [comment]
> This sounds awesome - especially because the folder-specific settings has been another idea i had.

I've been saving the per Folder settings in a `.gitjournal.yaml` file. But I'm not convinced it's a great idea as while experimenting with views it results in many commits. Also maybe not all users want this config stored in the git repo.

**Comment by mipro98** [comment]
I see the advantage of having a dedicated file for folder-specific settings. But for now, I personally don't see the necessity (yet). Even if the app gets reinstalled these small adjustments can be made quickly. Though if the amount of settings will become more and maybe if you're working in a repo with very many folders it might come in handy.

I personally don't care too much for now - I would be fine either way.

**Comment by vHanda** [comment]
Saving them in the folders saves me from having to keep the settings in sync if a folder is renamed or moved. But yes, it barely takes any time to set them again.
